source/Ship.cpp

```diff
@@ -988,6 +1008,22 @@ void Ship::Save(DataWriter &out) const
 			if(it.second)
 				out.Write("final explode", it.first->Name(), it.second);
 		});
+		if(hasHitEffect)
```

Comment:
No need to check if there are hit effects. If there are none the list is empty and nothing gets written.